pkg/operator/operator.go

```diff
@@ -496,12 +505,12 @@ func (o *Operator) sync(key string) error {
 	return nil
 }
 
-func (o *Operator) reportError(err error, taskName string) {
+func (o *Operator) reportError(err error, failedTaskReason string) {
 	klog.Infof("ClusterOperator reconciliation failed (attempt %d), retrying. Err: %v", o.failedReconcileAttempts+1, err)
 	if o.failedReconcileAttempts == 2 {
```

Comment:
side-note: once we declare failed, we might need to update the reason/message fields on every sync, otherwise the they wouldn't reflect the latest state. @jan--f WDYT?

```suggestion
	if o.failedReconcileAttempts >= 2 {
```

Reply:
@simonpasquier why does this need to be >=? 
Should the value be reset to 0 inside this condition so that it will report only after 3 attempts again?

Reply:
My take is that once the sync loop has failed for more than 3 times, we want to report all subsequent failures.

Reply:
> side-note: once we declare failed, we might need to update the reason/message fields on every sync, otherwise the they wouldn't reflect the latest state. @jan--f WDYT?

Yeah I think that's a great idea. Ideally we'd only update the reason if it changed. However that would require passing `failedReconcileAttempts` to `SetFailed`. There we could make the decision whether to update the reason or not. Not sure that is worth the extra complexity though when we could just blindly update once we have made more then 3 attempts.